dap-mode.el

```diff
@@ -1122,6 +1138,7 @@ arguments which contain the debug port to use for opening TCP connection."
   "Register configuration template CONFIGURATION-NAME.
 
 CONFIGURATION-SETTINGS - plist containing the preset settings for the configuration."
+  (dap--remove-debug-template configuration-name t)
```

Comment:
Why not smt like https://stackoverflow.com/questions/9812393/elisp-how-to-delete-an-element-from-an-association-list-with-string-key ?

Reply:
It's just because I'm still new with lisp and don't know this one yet :).
Will be adapted in next PS. Anyway, it only remove one item a time, right? So we still need dap--remove-debug-template.

Reply:
Can we have more than one match assuming that we clear the matching record just before inserting a new one? We may consider changing this to hash-table which will remove the need from this code.

Reply:
>Can we have more than one match

No for sure. It was stuck in my mind that we're living with a mixing configuration. Tkx for the good catch.
> changing this to hash-table

Yeah, this should be the way to go from beginning. Changing it now need much more adaption so I guess we can live with your suggestion.

Reply:
Yeah, back then when I was writing this code I was still learning elisp.